src/Symfony/Component/DependencyInjection/Compiler/AutowirePass.php

```diff
@@ -309,6 +319,10 @@ private function getReflectionClass($id, Definition $definition)
             $reflector = false;
         }
 
+        if ($deprecated) {
+            restore_error_handler();
```

Comment:
this needs to happen even when the previous code throws (eg the previous error handler)